### PR TITLE
[SPARK-47597][FOLLOWUP] Streaming: Migrate logInfo with variables to structured logging framework

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr is following up https://github.com/apache/spark/pull/46192

### Why are the changes needed?
When I reviewed the spark code, I found that some logs in the `streaming` module were not fully migrated to the structured logging framework, let's to complete unfinished work.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by panbingkun** [comment]
In addition, I found that other modules have similar issues. In another PR https://github.com/apache/spark/pull/46986, I did similar actions, some were omitted, and some were added but did not comply with the new structured log specifications.

File: streaming/src/main/scala/org/apache/spark/streaming/rdd/WriteAheadLogBackedBlockRDD.scala [hunk 1]
```diff
@@ -156,8 +157,8 @@ class WriteAheadLogBackedBlockRDD[T: ClassTag](
           s"Could not read data from write ahead log record ${partition.walRecordHandle}, " +
             s"read returned null")
       }
-      logInfo(s"Read partition data of $this from write ahead log, record handle " +
-        partition.walRecordHandle)
+      logInfo(log"Read partition data of ${MDC(LogKeys.RDD, this)} from write ahead log, " +
```
Comment (panbingkun) [inline streaming/src/main/scala/org/apache/spark/streaming/rdd/WriteAheadLogBackedBlockRDD.scala:160]: I'm not sure if it's appropriate to call it `LogKeys.RDD`

File: core/src/main/scala/org/apache/spark/executor/Executor.scala [hunk 2]
```diff
@@ -83,7 +83,7 @@ private[spark] class Executor(
   extends Logging {
 
   logInfo(log"Starting executor ID ${LogMDC(LogKeys.EXECUTOR_ID, executorId)}" +
-    log" on host ${LogMDC(HOST_NAME, executorHostname)}")
+    log" on host ${LogMDC(HOST, executorHostname)}")
```
Comment (panbingkun) [inline core/src/main/scala/org/apache/spark/executor/Executor.scala:86]: Let's call it `HOST` and remove `HOST_NAME`  from `LogKeys` at the same time.

File: core/src/main/scala/org/apache/spark/deploy/master/ui/MasterWebUI.scala [hunk 3]
```diff
@@ -79,7 +79,7 @@ class MasterWebUI(
           } else {
             val removedWorkers = masterEndpointRef.askSync[Integer](
               DecommissionWorkersOnHosts(hostnames))
-            logInfo(log"Decommissioning of hosts ${MDC(HOST_NAMES, hostnames)}" +
+            logInfo(log"Decommissioning of hosts ${MDC(HOSTS, hostnames)}" +
```
Comment (panbingkun) [inline core/src/main/scala/org/apache/spark/deploy/master/ui/MasterWebUI.scala:82]: Let's call it `HOSTS` and remove `HOST_NAMES` from `LogKeys` at the same time.

File: streaming/src/main/scala/org/apache/spark/streaming/scheduler/ExecutorAllocationManager.scala [hunk 4]
```diff
@@ -75,8 +75,10 @@ private[streaming] class ExecutorAllocationManager(
 
   def start(): Unit = {
     timer.start()
-    logInfo(s"ExecutorAllocationManager started with " +
-      s"ratios = [$scalingUpRatio, $scalingDownRatio] and interval = $scalingIntervalSecs sec")
+    logInfo(log"ExecutorAllocationManager started with " +
+      log"ratios = [${MDC(LogKeys.SCALING_UP_RATIO, scalingUpRatio)}, " +
+      log"${MDC(LogKeys.SCALING_DOWN_RATIO, scalingDownRatio)}] and " +
+      log"interval = ${MDC(LogKeys.SCALING_INTERVAL, scalingIntervalSecs * 1000)} ms")
```
Comment (panbingkun) [inline streaming/src/main/scala/org/apache/spark/streaming/scheduler/ExecutorAllocationManager.scala:81]: interval -> Unified time unit as `ms`

File: streaming/src/main/scala/org/apache/spark/streaming/scheduler/ExecutorAllocationManager.scala [hunk 5]
```diff
@@ -89,11 +91,14 @@ private[streaming] class ExecutorAllocationManager(
...
    */
   private def manageAllocation(): Unit = synchronized {
-    logInfo(s"Managing executor allocation with ratios = [$scalingUpRatio, $scalingDownRatio]")
+    logInfo(log"Managing executor allocation with ratios = [" +
+      log"${MDC(LogKeys.SCALING_UP_RATIO, scalingUpRatio)}, " +
+      log"${MDC(LogKeys.SCALING_DOWN_RATIO, scalingDownRatio)}]")
     if (batchProcTimeCount > 0) {
       val averageBatchProcTime = batchProcTimeSum / batchProcTimeCount
       val ratio = averageBatchProcTime.toDouble / batchDurationMs
-      logInfo(s"Average: $averageBatchProcTime, ratio = $ratio" )
```
Comment (panbingkun) [inline streaming/src/main/scala/org/apache/spark/streaming/scheduler/ExecutorAllocationManager.scala:96]: Remove redundant `whitespaces`

File: streaming/src/main/scala/org/apache/spark/streaming/scheduler/ExecutorAllocationManager.scala [hunk 6]
```diff
@@ -142,9 +149,9 @@ private[streaming] class ExecutorAllocationManager(
         } else {
           client.killExecutor(execIdToRemove)
         }
-        logInfo(s"Requested to kill executor $execIdToRemove")
```
Comment (panbingkun) [inline streaming/src/main/scala/org/apache/spark/streaming/scheduler/ExecutorAllocationManager.scala:145]: Remove redundant `interpolation`

File: streaming/src/main/scala/org/apache/spark/streaming/scheduler/ReceiverTracker.scala [hunk 7]
```diff
@@ -232,7 +232,7 @@ class ReceiverTracker(ssc: StreamingContext, skipReceiverLaunch: Boolean = false
 
     // Signal the receivers to delete old block data
     if (WriteAheadLogUtils.enableReceiverLog(ssc.conf)) {
-      logInfo(s"Cleanup old received batch data: $cleanupThreshTime")
+      logInfo(log"Cleanup old received batch data: ${MDC(CLEANUP_THRESH_TIME, cleanupThreshTime)}")
```
Comment (panbingkun) [inline streaming/src/main/scala/org/apache/spark/streaming/scheduler/ReceiverTracker.scala:235]: `Time` itself is displayed in `ms`

File: streaming/src/main/scala/org/apache/spark/streaming/dstream/DStream.scala [hunk 8]
```diff
@@ -201,7 +200,8 @@ abstract class DStream[T: ClassTag] (
     // Set the checkpoint interval to be slideDuration or 10 seconds, which ever is larger
     if (mustCheckpoint && checkpointDuration == null) {
       checkpointDuration = slideDuration * math.ceil(Seconds(10) / slideDuration).toInt
-      logInfo(s"Checkpoint interval automatically set to $checkpointDuration")
+      logInfo(log"Checkpoint interval automatically set to " +
```
Comment (panbingkun) [inline streaming/src/main/scala/org/apache/spark/streaming/dstream/DStream.scala:203]: `Duration` itself is displayed in `ms`

**Comment by panbingkun** [comment]
cc @gengliangwang

**Comment by gengliangwang** [comment]
@panbingkun Thanks for the work. There is another pending PR created before this one: https://github.com/apache/spark/pull/46980
Let's hold this one until https://github.com/apache/spark/pull/46980 is merged.

**Comment by panbingkun** [comment]
> @panbingkun Thanks for the work. There is another pending PR created before this one: #46980 Let's hold this one until #46980 is merged.

Oh, okay, let's hold it first and turn it into a draft state first.

**Pull request closed without merging**